Tasks/AzurePowerShellV4/TryMakingModuleAvailable.ps1

```diff
@@ -0,0 +1,81 @@
+[CmdletBinding()]
+param (
+    [string]
+    $targetVersion,
+
+    [string] [Parameter(Mandatory = $true)] [ValidateSet("Windows", "Linux", "Mac")]
+    $platform
+)
+
+try {
+    $isWin = $platform -eq "Windows"
+
+    . (Join-Path $PSScriptRoot "Utility.ps1")
+    $isHostedAgent = Test-IsHostedAgentPathPresent -isWin $isWin
+ 
+    if (!$isHostedAgent) {
+        Write-Verbose "Module path not present as expected in hosted agent, skipping step to make module available."
+        $moduleSource = "Others"
+        return
+    }
+
+    if (!$targetVersion) {
+        Write-Verbose "Latest selected, will make use of the latest available in agent as folder."
+        $moduleSource = "Folder"
+        return
+    }
+
+    $modulePath = Get-SavedModulePath -azurePowerShellVersion $targetVersion -isWin $isWin
+    if (Test-Path $modulePath) {
+        Write-Verbose "Az $targetVersion present at $modulePath as folder."
+        $moduleSource = "Folder"
+        return
+    }
+
+    $moduleContainerPath = Get-SavedModuleContainerPath -isWin $isWin
+    $moduleZipPath = $modulePath + ".zip";
+    if (Test-Path $moduleZipPath) {
+        Write-Verbose "Az $targetVersion present at $moduleZipPath as zip, expanding it."
+        Expand-ModuleZip -zipPath $moduleZipPath -destination $moduleContainerPath -isWin $isWin
+        Write-Verbose "Zip expanded"
+        $moduleSource = "Zip"
+        return
+    }
+
+    try {
+        Write-Verbose "Getting versions manifest from GHRelease."
+        $versionsManifest = Invoke-RestMethod -Method Get `
+            -Headers @{ "Accept" = "application/vnd.github.VERSION.raw" } `
+            -Uri "https://api.github.com/repos/Azure/az-ps-module-versions/contents/versions-manifest.json"
+        $downloadUrlEntity = $versionsManifest | Where-Object version -eq $targetVersion
```

Comment:
Log versions manifest and a log stating retrieval succeeded.

Reply:
We are not able to log versions manifest content since `write-verbose` expects string type while `Invoke-RestMethod` call will get us an `System.Object[]` which is creating issue in converting to string.